examples/ex01_inputfile/src/main.C

```diff
@@ -16,9 +16,8 @@
 #include "ExampleApp.h"
 
 // Moose Includes
-#include "MooseInit.h"
-#include "MooseApp.h"
-#include "AppFactory.h"
+#include "Moose.h"
+#include "MooseMain.h"
 
 // Create a performance log
 PerfLog Moose::perf_log("Example");
```

Comment:
@roystgnr can we move this around?

Reply:
I moved `perf_log()` into `main()`, it compiles fine but failed all the tests as below. Does this a registration error?
`symbol not found in flat namespace '__ZN5Moose8perf_logE'`
Here is where I moved `perf_log()`, did I do anything wrong?
```
template <typename DefaultAppType>
void
main(int argc, char * argv[])
{
  MooseInit init(argc, argv);

  DefaultAppType::registerApps();

  const auto default_app_name = MooseUtils::prettyCppType<DefaultAppType>();

  // Create a performance log
  PerfLog perf_log(default_app_name);

  auto app = createMooseApp(default_app_name, argc, argv);

  app->run();
}
```

Reply:
Sorry, missed the notification somehow.

That linker error looks like the definition of `perf_log` has been moved but there's still a *declaration* of `perf_log` (possibly one being used!) which hasn't.